lib/adapters/show-document-adapter.ts

```diff
@@ -0,0 +1,52 @@
+import { shell } from "electron"
+import { LanguageClientConnection, ShowDocumentParams, ShowDocumentResult } from "../languageclient"
+import { TextEditor } from "atom"
+import Convert from "../convert"
+
+/** Public: Adapts the window/showDocument command to Atom's text editors or external programs. */
+export class ShowDocumentAdapter {
```

Comment:
What is the reason we export a class and the functions separately?

Reply:
Can we just make the functions static like other adapters.

Reply:
I added the class just in case someone wants to overload them. Otherwise, I would remove it.

Reply:
I think I can remove the class. Since people can call the functions directly. There is no point in adding class bloat here.

Reply:
We should probably keep it consistent with the other adapters. I'm ok with removing the class if that would work for the others. (Probably in another PR)

Reply:
After looking at the other adapters it seems like exporting the class with static functions would be better to keep importing adapters consistent. Otherwise it will be a pain figuring out which adapters need `import * as adapter from 'adapter'` and which ones need `import adapter from 'adapter'`. It would be better to only import on way.